core/save-profiledata.cpp

```diff
@@ -45,6 +47,16 @@ static std::string video_time(int secs)
 	return format_string_std("%d:%02d:%02d.000,", hours, mins, secs);
 }
 
+void replace_all(std::string& str, const std::string& old_value, const std::string& new_value) {
+    if (old_value.empty())
```

Comment:
This check seems unnecessary to me. If `old_value` is empty, the loop body will simply not execute. The common case will be a non-empty string, so no point in wasting code on this.

Reply:
If `old_value` is empty, it will always match at `start_pos`, resulting in it getting stuck in an infinite loop.

Reply:
Yes correct - I thought this was testing the modified string.